horovod/run/run.py

```diff
@@ -308,9 +324,14 @@ def run():
     # --disable-cache flag.
     fn_cache = None
     if not args.disable_cache:
-        parameters_hash = hashlib.md5((' '.join([str(args.host),
-                                                 str(args.np),
-                                                 str(args.ssh_port)]))).hexdigest()
+        params = ''
+        if args.np:
+            params += str(args.np)
```

Comment:
Add separators?  Otherwise, you can have `host=host12345`, which will collide with `host=host` and `port=12345`.  Not very likely, but it's a good practice to think about such collisions.

Reply:
added.